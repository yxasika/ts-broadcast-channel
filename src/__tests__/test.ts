import { BroadcasteOnChannel, clearBroadcastChannelByName } from '../index';

const __testChannels = {
  __numberChannel: 'some-number-channel',
  __voidChannel: 'some-void-channel',
};

class __TestService {
  public __voidWasCalled: boolean = false;

  @BroadcasteOnChannel({ channel: __testChannels.__voidChannel })
  private someBroadcastedMethod(): void {
    this.__voidWasCalled = true;
    return;
  }

  public __numberWasCalled: boolean = false;

  @BroadcasteOnChannel({ channel: __testChannels.__numberChannel })
  private someBroadcastedFunction(a: number, b: number): number {
    return a + b;
  }
}

describe('Broadcasted', () => {
  let testee: __TestService;

  beforeEach(() => (testee = new __TestService()));

  test('should not change prototype of class', () => {
    expect(Object.getPrototypeOf(testee)).toMatchSnapshot();
  });

  afterEach(() => {
    clearBroadcastChannelByName(__testChannels.__voidChannel);
    clearBroadcastChannelByName(__testChannels.__numberChannel);
  });
});
