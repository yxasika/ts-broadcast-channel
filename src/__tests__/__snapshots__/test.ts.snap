// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Broadcasted should not change prototype of class 1`] = `
__TestService {
  "someBroadcastedFunction": [Function],
  "someBroadcastedMethod": [Function],
}
`;
